pennylane/measurements/counts.py

```diff
@@ -135,7 +140,125 @@ def circuit(x):
             )
         wires = Wires(wires)
 
+    # TODO: Remove this conditional branch when using `Counts.process` in devices
     if all_outcomes:
-        return MeasurementProcess(AllCounts, obs=op, wires=wires)
-
-    return MeasurementProcess(Counts, obs=op, wires=wires)
+        return _Counts(AllCounts, obs=op, wires=wires, all_outcomes=all_outcomes)
+
+    return _Counts(Counts, obs=op, wires=wires, all_outcomes=all_outcomes)
+
+
+# TODO: Make public when removing the ObservableReturnTypes enum
+class _Counts(SampleMeasurement):
+    """Measurement process that returns the samples of a given observable."""
+
+    def __init__(
+        self,
+        return_type: ObservableReturnTypes,
+        obs: Union[Observable, None] = None,
+        wires=None,
+        eigvals=None,
+        id=None,
+        all_outcomes=False,
+    ):
+        self.all_outcomes = all_outcomes
+        super().__init__(return_type, obs, wires, eigvals, id)
+
+    def process_samples(
+        self,
+        samples: Sequence[complex],
+        wire_order: Wires,
+        shot_range: Tuple[int] = None,
+        bin_size: int = None,
+    ):
+        samples = qml.sample(op=self.obs, wires=self._wires).process_samples(
+            samples, wire_order, shot_range, bin_size
+        )
```

Comment:
I started doing this, but then I realised that there are some measurements that use the logic of multiple measurements (for example `Variance` uses the logic of `Sample` and `Probability`). I decided to do it this way to avoid having `super().super()` calls.

Reply:
Alright, fair.